pkg/storage/legacysql/dualwrite/dualwriter.go

```diff
@@ -29,35 +30,58 @@ type dualWriter struct {
 }
 
 func (d *dualWriter) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
-	// Call get (send read traffic in cloud)
-	unifiedGet, unifiedErr := d.unified.Get(ctx, name, options)
+	// If we read from unified, we can just do that and return.
 	if d.readUnified {
-		return unifiedGet, unifiedErr
+		return d.unified.Get(ctx, name, options)
 	}
-
+	// If legacy is still our main store, lets first read from it.
 	legacyGet, err := d.legacy.Get(ctx, name, options)
 	if err != nil {
 		return nil, err
 	}
-
-	if unifiedErr != nil && !apierrors.IsNotFound(unifiedErr) && !d.errorIsOK {
-		return nil, unifiedErr // the unified error
+	// Once we have successfully read from legacy, we can check if we want to fail on a unified read.
+	// If we allow the unified read to fail, we can do it in the background.
+	if d.errorIsOK {
+		go func() {
+			if _, err := d.unified.Get(ctx, name, options); err != nil {
```

Comment:
@JohnnyQQQQ i think these need background context

Reply:
scratch that, @stephaniehingtgen pointed me to https://github.com/grafana/grafana/pull/102518. it looks like it hadn't deployed yet so i was seeing traces with cancellations